one-ui/ui/src/components/entities/mapping/mapping-card.tsx

```diff
@@ -103,6 +110,142 @@ const MappingCard: React.FC<Props> = (props) => {
+            setDocNotFound(true);
+        }
+           
+       
+    }
+
+    const fetchSrcDocFromUri = async (uri) => {
+        try{
+            let srcDocResp = await getDoc('STAGING', uri);
+            if (srcDocResp.status === 200) {
+                let nestedDoc: any = [];
+                let docRoot = srcDocResp.data['envelope'] ? srcDocResp.data['envelope']['instance'] : srcDocResp.data;
+                let sDta = generateNestedDataSource(docRoot,nestedDoc);
+                //setSourceData(prevState => ([ ...prevState, ...sDta]));
+                setSourceData([...sDta]);
+            }
+            } catch(error)  {
+                let message = error.response.data.message;
+                console.log('Error While loading the Doc from URI!', message)
+                setDocNotFound(true);
+            }
+    }
+
+
+    // construct infinitely nested source Data
+    const generateNestedDataSource = (respData, nestedDoc: Array<any>) => {
+        
+        Object.keys(respData).map(key => {
+            let val = respData[key];
+            if (val != null && val!= "") {
+   
+                if (val.constructor.name === "Object") {
+     
+                    let propty = {
+                        key: key,
+                        'children': []
+                    }
+
+                    generateNestedDataSource(val, propty.children);
+                    nestedDoc.push(propty);
+
+                } else if (val.constructor.name === "Array") {
+                    //srcData.push({key : key, val: respData[key]})
+                    
+                    val.forEach(obj => {
+                        if(obj.constructor.name == "String"){
+                          let propty = {
+                            key: key,
+                            val: obj
+                          };
+                          nestedDoc.push(propty);
+                        } else {
+                            let propty = {
+                                key: key,
+                                children: []
+                              };
+                              
+                          generateNestedDataSource(obj, propty.children);
+                          nestedDoc.push(propty);
+                        }
+                      });
+
+                } else {
+
+                    let propty = {
+                        key: key,
+                        val: String(val)
+                      };
+                    nestedDoc.push(propty);
+                }
+
+            } else {
+
+                let propty = {
+                    key: key,
+                    val: ""
+                  };
+                nestedDoc.push(propty);
+            }
+        });
+
+        return nestedDoc;
+        
+        
+        
+    }
+    
+    
+    const extractEntityInfoForTable = () => {
+        let entProps = props.entitiesInfo.definitions.definitions[props.entityName].properties;
```

Comment:
I recommend "entityModel" instead of "entitiesInfo" (though why are there two levels of "definitions" here?) and then "entityTypeTitle" instead of "entityName".

Reply:
Done.